### PR TITLE
[REVIEW][FEATURE][UWANG-161] - integration statistic card

### PR DESCRIPTION
**Pull request by maulanafikriash**
#### Description

> Description of features, improvements design.

#### Ticket Reference(s)

[UWANG-161](https://bluehabit.atlassian.net/browse/UWANG-161)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> Mention other user or extra note unrelated to the MR here.
@triandamai 

[UWANG-161]: https://bluehabit.atlassian.net/browse/UWANG-161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/StatisticCard.vue [hunk 1]
```diff
@@ -27,7 +27,6 @@ function isStatisticDown(statistic:number) {
   console.log("up",statistic);
```
Comment (triandamai) [inline components/StatisticCard.vue]: hapus ya @maulanafikriash

File: pages/preview.vue [hunk 2]
```diff
@@ -1,22 +1,30 @@
 <script setup lang="ts">
+import { useUserStatisticsStore } from '~/store/getStatisticCard';
```
Comment (triandamai) [inline pages/preview.vue]: gaperlu import @maulanafikriash nuxt udah otomatis import

File: store/getStatisticCard.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      user_claim: '',
+      total_user: '',
+    },
+    isLoading: true,
+  }),
+  actions: {
+    async fetchStatistics() {
+      this.isLoading = true;
+      try {
+        const response = await fetch('https://api.bluhabit.id/account/v1/admin/get-statistic');
```
Comment (triandamai) [inline store/getStatisticCard.ts]: pake `useApi` ya @maulanafikriash

File: store/getStatisticCard.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        const data = await response.json();
+        this.statistics = data.data;
+      } catch (error) {
+        console.error('Error fetching statistics:', error);
```
Comment (triandamai) [inline store/getStatisticCard.ts]: remove ya @maulanafikriash

**Review by triandamai — CHANGES_REQUESTED** [review]
update dulu ya

File: store/getStatisticCard.ts (hunk 4 above)
Comment (maulanafikriash) [reply store/getStatisticCard.ts]: oke sudah mas @triandamai

File: store/getStatisticCard.ts [hunk 5]
```diff
@@ -1,19 +1,26 @@
...
     isLoading: true,
   }),
   actions: {
     async fetchStatistics() {
       this.isLoading = true;
       try {
-        const response = await fetch('https://api.bluhabit.id/account/v1/admin/get-statistic');
-        const data = await response.json();
-        this.statistics = data.data;
+        const api = useApi();
```
Comment (triandamai) [inline store/getStatisticCard.ts]:
```javascript
const {get} = useApi()
const data = await get<StatisticResponse>("/<url>")
```
ps: `StatisticResponse` bikin di file `type.ts`, `url` cukup endpoint aja `https//.api.bluhabit.id` nya gausah

File: store/getStatisticCard.ts [hunk 6]
```diff
@@ -1,19 +1,26 @@
+interface StatisticData {
```
Comment (triandamai) [inline store/getStatisticCard.ts]: pindah ke `type.ts`

**Review by triandamai — CHANGES_REQUESTED** [review]
cek ya

File: store/getStatisticCard.ts [hunk 7]
```diff
@@ -1,26 +1,24 @@
-interface StatisticData {
-  user_active: string;
-  user_claim: string;
-  total_user: string;
-}
+import type { StatisticData } from "~/type/statisticData";
+import type { StatisticResponse } from "~/type/statisticResponse";
 
 export const useUserStatisticsStore = defineStore('getStatisticsCard', {
```
Comment (triandamai) [inline store/getStatisticCard.ts]: `defineStore<string,{statistics:StatisticData,isLoading:boolean}>(...)`

**Review by triandamai — CHANGES_REQUESTED** [review]
cek

File: store/getStatisticCard.ts (hunk 7 above)
Comment (triandamai) [reply store/getStatisticCard.ts]: export const useStatistic = defineStore("statistic",....

Comment (maulanafikriash) [reply store/getStatisticCard.ts]: dah ku ubah, dicek lagi mas @triandamai

File: store/getStatisticCard.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import type { StatisticData } from "~/type/statisticData";
```
Comment (triandamai) [inline store/getStatisticCard.ts]: nama store nya ganti `useStatistic` @maulanafikriash

File: store/getStatisticCard.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+      total_user: 0,
+    } as StatisticData,
+    isLoading: true,
+  }),
+  actions: {
+    async fetchStatistics() {
+      this.isLoading = true;
+      try {
+        const { get } = useApi();
+        const { data } = await get<StatisticResponse>('https://api.bluhabit.id/uwang/dev/account/v1/admin/get-statistic');
```
Comment (triandamai) [inline store/getStatisticCard.ts]: ga perlu full url ya kan udah di set base urlnya @maulanafikriash

File: store/getStatisticCard.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import type { StatisticData } from "~/type/statisticData";
+import type { StatisticResponse } from "~/type/statisticResponse";
+
+export const useStatistic = defineStore('statistic', {
+  state: () => ({
```
Comment (triandamai) [inline store/getStatisticCard.ts]: kalo mau ada type nya gausah pake `as` tapi:
```typescript
const useStatistic = defineStore<string,{/* ini type state*/},{/* ini type buat getter */},{/* ini type action */}>('statistic',{
    state:()=>({})// ini bakal sesuai dengan type yang uda di define
})
```

File: store/getStatisticCard.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    statistics: {
+      active_user: 0,
```
Comment (triandamai) [inline store/getStatisticCard.ts]: pake `camelCase` ya `snake_case` hanya di pakai untuk pertukaran data api

File: store/getStatisticCard.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      user_claim: 0,
+      total_user: 0,
+    } as StatisticData,
+    isLoading: true,
+  }),
+  actions: {
+    async fetchStatistics() {
+      this.isLoading = true;
+      try {
```
Comment (triandamai) [inline store/getStatisticCard.ts]: gausah pake try catch karena bakal susah debugging kalo ada error

**Review by triandamai — CHANGES_REQUESTED** [review]
revisi dikit @maulanafikriash

File: store/getStatisticCard.ts (hunk 12 above)
Comment (maulanafikriash) [reply store/getStatisticCard.ts]: OKe dah ku ubah mas @triandamai

File: pages/preview.vue [hunk 13]
```diff
@@ -1,5 +1,5 @@
 <script setup lang="ts">
-import { useStatistic } from '~/store/getStatisticCard';
+import { useStatistic } from '~/store/useStatistic';
```
Comment (triandamai) [inline pages/preview.vue]: ga perlu import ya @maulanafikriash udah otomatis dari nuxt

**Review by triandamai — CHANGES_REQUESTED** [review]
last @maulanafikriash

File: pages/preview.vue (hunk 13 above)
Comment (maulanafikriash) [reply pages/preview.vue]: error tapi mas, apa biarin aja ? @triandamai

Comment (triandamai) [reply pages/preview.vue]: itu warning dari intellisense aja

Comment (maulanafikriash) [reply pages/preview.vue]: oke sip mas @triandamai

Comment (maulanafikriash) [reply pages/preview.vue]: dah ku remove mas importnya @triandamai

**Review by triandamai — CHANGES_REQUESTED** [review]
di integrasiin ke pages/dashboard ya

**Comment by triandamai** [comment]
@maulanafikriash

**Comment by maulanafikriash** [comment]
Oke udah mas di integrasiin ke pages/main/dashboard @triandamai

**Pull request merged**